data-pipeline/src/covid_variant_pipeline/assign_clades.py

```diff
@@ -123,14 +126,27 @@ def assign_clades(reference_tree: str):
     return assignment_file
 
 
-def main():
-    logger.info("Starting pipeline")
+@click.command()
+@click.option(
+    "--as-of-date",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
```

Comment:
setting this parameter as DateTime type is just a lazy way of validating the input